x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/LdapUserSearchSessionFactory.java

```diff
@@ -66,14 +67,14 @@ class LdapUserSearchSessionFactory extends PoolingSessionFactory {
         logger.info("Realm [{}] is in user-search mode - base_dn=[{}], search filter=[{}]", config.name(), userSearchBaseDn, searchFilter);
     }
 
-    static boolean hasUserSearchSettings(RealmConfig config) {
+    static List<? extends Setting.AffixSetting<?>> configuredUserSearchSettings(RealmConfig config) {
         return Stream.of(
             LdapUserSearchSessionFactorySettings.SEARCH_BASE_DN,
             LdapUserSearchSessionFactorySettings.SEARCH_ATTRIBUTE,
             LdapUserSearchSessionFactorySettings.SEARCH_SCOPE,
             LdapUserSearchSessionFactorySettings.SEARCH_FILTER,
             LdapUserSearchSessionFactorySettings.POOL_ENABLED
-        ).anyMatch(config::hasSetting);
+        ).filter(config::hasSetting).toList();
```

Comment:
Theorectically all settings under `user_search.` prefix should be considered. But that would be a breaking change. So keep this logic as is with these 5 settings.